src/Line.cpp

```diff
@@ -111,15 +102,16 @@ bool Line::doesIntersect(const LineSegment &line) const {
     return intersects(line).has_value();
 }
 
-std::optional<std::pair<Vector2, float>> Line::generalIntersect(const Line &line) const {
-    Vector2 A = start - end;
-    Vector2 B = line.start - line.end;
+std::optional<std::tuple<Vector2, float, float>> Line::generalIntersect(const Vector2 line1Start, const Vector2 line1End, const Vector2 line2Start, const Vector2 line2End) {
+    Vector2 A = line1Start - line1End;
+    Vector2 B = line2Start - line2End;
     double denom = A.cross(B);
     if (denom != 0) {
-        Vector2 C = start - line.start;
-        double numer = C.cross(A);
-        double u = numer / denom;
-        return std::pair(line.start - B*u, u);
+        Vector2 C = line1Start - line2Start;
+        double numer = C.cross(B);
+        double t = numer / denom;
+        double u = -A.cross(C) / denom;
+        return std::tuple(line1Start - A*t, t, u);
     }
     return std::nullopt;
```

Comment:
Don't change this function. It was working fine, and you are giving it responsibility it should not have